src/huggingface_hub/inference/_common.py

```diff
@@ -265,34 +265,44 @@ def _bytes_to_image(content: bytes) -> "Image":
 ## STREAMING UTILS
 
 
-def _stream_text_generation_response(
+def _stream_text_generation_response_from_bytes(
     bytes_output_as_lines: Iterable[bytes], details: bool
 ) -> Union[Iterable[str], Iterable[TextGenerationStreamOutput]]:
     """Used in `InferenceClient.text_generation`."""
     # Parse ServerSentEvents
     for byte_payload in bytes_output_as_lines:
-        output = _format_text_generation_stream_output(byte_payload, details)
+        try:
+            output = _format_text_generation_stream_output(byte_payload, details)
+        except StopIteration:
+            break
         if output is not None:
             yield output
 
 
-async def _async_stream_text_generation_response(
+async def _async_stream_text_generation_response_from_bytes(
```

Comment:
I'm not sure why this warranted a rename but ok.

(the current name feels just way too long to me).

Reply:
removed `_from_bytes` everywhere